structuralcodes/codes/mc2010/_concrete_punching.py

```diff
@@ -0,0 +1,579 @@
+        l_y (float): The width in x direction that the collumn carries
+        l_min (float): The shorter side of the the L_x and L_y
+        inner (bool): Is true only if the column is a inner column
+        edge_par (bool): Is true only if the column is a edge column with
+        tention reinforcement paralell to the edge
+        edge_per (bool): Is true only if the column is a edge column with
+        tention reinforcement perpendicular to the edge
+        corner (bool): Is true only if the column is a corner column
+
+    Return:
+        The bending moment acting in the support strip regardless of the
+        position of the column
+    """
+    r_sx = 0.22 * l_x
+    r_sy = 0.22 * l_y
+    b_s = min(1.5 * (r_sx * r_sy) ** 0.5, l_min)
+    if inner:
+        return v_ed * ((1 / 8) + e_u / (2 * b_s))
+    if edge_par:
+        return max(v_ed * ((1 / 8) + e_u / (2 * b_s)), v_ed / 4)
+    if edge_per:
+        return v_ed * ((1 / 8) + e_u / (b_s))
+    if corner:
+        return max(v_ed * ((1 / 8) + e_u / (b_s)), v_ed / 2)
+    raise ValueError("Placement is not defined, only one needs to be True")
+
+
+def psi_punching(
+    l_x: float,
+    l_y: float,
+    f_yd: float,
+    d: float,
+    e_s: float,
+    approx_lvl_p: float,
+    v_ed: float,
+    e_u: float,
+    l_min: float,
+    inner: bool,
+    edge_par: bool,
+    edge_per: bool,
+    corner: bool,
+    m_rd: float,
+    m_pd: float,
+) -> float:
+    """The rotation of the slab around the supported area
+
+    fib Model Code 2010, eq. (7.3-70), (7.3-75) and (7.3-77)
+
+    Args:
+        l_x (float): The distance between two columns in x direction
+        l_y (float): The distance between two columns in y direction
+        f_yd (float): Design strength of reinforment steel in MPa
+        d (float): The mean value of the effective depth in mm
+        e_s (float): The E_s-modulus for steel in Mpa
+        approx_lvl_p (float): The approx level for punching
+        v_ed (float): The acting shear force from the columns
+        e_u (float): Refers to the eccentricity of the resultant of shear
+        forces with respect to the centroid
+        l_min (float): The shorter side of the the L_x and L_y
+        inner (bool): Is true only if the column is a inner column
+        edge_par (bool): Is true only if the column is a edge column with
+        tention reinforcement paralell to the edge
+        edge_per (bool): Is true only if the column is a edge column with
+        tention reinforcement perpendicular to the edge
+        corner (bool): Is true only if the column is a corner column
+        m_rd (float): The design average strength per unit length in MPa
+        m_pd: (float): The average decompresstion moment due to prestressing
+        in MPa
+
+    Return:
+        psi for the choosen approx level in punching"""
+
+    r_s = max(0.22 * l_x, 0.22 * l_y)
+
+    if approx_lvl_p == 1:
+        if not 0.5 < l_x / l_y < 2:
```

Comment:
In LoA 1, the maximum value of `r_s` should be used, and this is calculated in line 115. No more checks seem to be necessary.